### PR TITLE
changes to support mathematical function expressions during query pushdown

### PR DESCRIPTION
**Pull request by suryasoma**
changes to support mathematical function expressions during query pushdown
Added unit tests and integration tests

### CODE REVIEW

File: spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/SparkExpressionConverter.scala [hunk 1]
```diff
@@ -41,6 +41,7 @@ trait SparkExpressionConverter {
       .orElse(convertBooleanExpressions(expression, fields))
       .orElse(convertMiscExpressions(expression, fields))
       .orElse(convertStringExpressions(expression, fields))
+      .orElse(convertMathematicalExpressions(expression, fields))
```
Comment (VinayLondhe14) [inline spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/SparkExpressionConverter.scala:44]: Minor: Can you move this to above `convertMiscExpressions` to preserve alphabetical order?

File: spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/QueryPushdownIntegrationTestBase.java [hunk 2]
```diff
@@ -61,4 +62,63 @@ public void testStringFunctionExpressions() {
...
+    assertThat(r1.get(6)).isEqualTo(BigDecimal.valueOf(23)); // CEIL(22.5)
+    assertThat(r1.get(7)).isEqualTo(1.0); // COS(0)
+    assertThat(r1.get(8)).isEqualTo(1.0); // COSH(0)
+    assertThat(r1.get(9)).isEqualTo(2.72); // ROUND(EXP(1),2)
+    assertThat(r1.get(10)).isEqualTo(2); // FLOOR(EXP(1))
+    assertThat(r1.get(11)).isEqualTo(5); // GREATEST(1,5,3,4)
+    assertThat(r1.get(12)).isEqualTo(1); // LEAST(1,5,3,4)
+    assertThat(r1.get(13)).isEqualTo(2.3); // ROUND(LOG(word_count), 2)
+    assertThat(r1.get(14)).isEqualTo(1.0); // ROUND(LOG10(word_count), 2)
+    assertThat(r1.get(15)).isEqualTo(100.0); // POW(word_count, 2)
```
Comment (VinayLondhe14) [inline spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/QueryPushdownIntegrationTestBase.java:114]: I dont see an assert for number 16

File: spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/SparkExpressionConverter.scala [hunk 3]
```diff
@@ -199,6 +200,25 @@ trait SparkExpressionConverter {
...
+      case _ => null
+    })
+  }
+
+  def convertMathematicalExpressions(expression: Expression, fields: Seq[Attribute]): Option[BigQuerySQLStatement] = {
+    Option(expression match {
+      case _: Abs | _: Acos | _: Asin | _: Atan | _: Ceil |
+           _: Cos | _: Cosh | _: Exp | _: Floor | _: Greatest |
+           _: Least | _: Logarithm | _: Log | _: Log10 |
+           _: Pow | _:Round | _: Sin | _: Sinh |
```
Comment (VinayLondhe14) [inline spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/SparkExpressionConverter.scala:213]: Does the Round expression here also take care when scale is given in the input? i.e. Does it take care of 
`case Round(child, scale)`

Comment (suryasoma) [reply spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/SparkExpressionConverter.scala:213]: Yeah it does Vinay. I have used `Round(child, scale)` function for many functions in the integration tests.
Like: `ROUND(LOG(word_count), 2)` `ROUND(ATAN(0.5),2)`

File: spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/SparkExpressionConverter.scala (hunk 1 above)
Comment (suryasoma) [reply spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/SparkExpressionConverter.scala:44]: Yeah Sure

**Review by VinayLondhe14 — APPROVED** [review]

**Pull request merged**